eng/tools/rush-runner.js

```diff
@@ -99,7 +178,11 @@ if (serviceDirs.length === 0) {
       break;
 
     default:
-      params = flatMap(packageNames, (p) => [`--to`, p]);
+      let requiredPackageNames = packageNames;
+      if (buildTransitiveDep) {
+        requiredPackageNames = getPackagesToBuild(packageNames, pkgGraph);
+      }
+      params = flatMap(requiredPackageNames, (p) => [`--to`, p]);
```

Comment:
Do we need to do anything to ensure we build things in the correct order?

Reply:
As far I know, rush is taking care of building it in order. So finding the order in our side may be overkill since we don't find all required chain. We pass only leaf projects to rush so rush can find all required projects and order to build.